lib/SILOptimizer/FunctionSignatureTransforms/ExistentialSpecializer.cpp

```diff
@@ -64,12 +74,50 @@ class ExistentialSpecializer : public SILFunctionTransform {
     /// Get CallerAnalysis information handy.
     CA = PM->getAnalysis<CallerAnalysis>();
 
+    /// Get ProtocolConformanceAnalysis.
+    PCA = PM->getAnalysis<ProtocolConformanceAnalysis>();
+
+    /// Get ClassHierarchyAnalysis.
+    CHA = PM->getAnalysis<ClassHierarchyAnalysis>();
     /// Perform specialization.
     specializeExistentialArgsInAppliesWithinFunction(*F);
   }
 };
 } // namespace
 
+/// Find concrete type from Sole Conforming Type.
+bool ExistentialSpecializer::findConcreteTypeFromSoleConformingType(
```

Comment:
This method checks if a SoleConformingType can be determined for the argument. Some parts of code (5-6 lines) overlap with what was done in SILCombiner, but there are lot of disparities to make it a new APi in Existential.h altogether.

Reply:
@atrick You had a similar concern from SILCombiner, but the overlap is minor to warrant a new API?